apps/air-discount-scheme/backend/src/app/modules/discount/discount.module.ts

```diff
@@ -1,12 +1,17 @@
-import { Module } from '@nestjs/common'
+import { forwardRef, Module } from '@nestjs/common'
 
 import { PrivateDiscountController } from './discount.controller'
 import { DiscountService } from './discount.service'
 import { CacheModule } from '../cache'
 import { NationalRegistryModule } from '../nationalRegistry'
+import { FlightModule } from '../flight'
 
 @Module({
-  imports: [CacheModule, NationalRegistryModule],
+  imports: [
+    CacheModule,
+    NationalRegistryModule,
+    forwardRef(() => FlightModule),
```

Comment:
Where I commented above, import he module instead. I see that you're already doing it.. so you don't need the `Inject` decorator if the flightService provider is correctly exported.